sessionctx/variable/session.go

```diff
@@ -94,18 +94,15 @@ func (s *SessionVars) AddFoundRows(rows uint64) {
 	s.FoundRows += rows
 }
 
-// SetStatus sets the session server status variable
-func (s *SessionVars) SetStatus(status uint16) {
-	s.Status = status
-}
-
-// SetStatusInTrans sets the status flags about ServerStatusInTrans.
-func (s *SessionVars) SetStatusInTrans(isInTrans bool) {
-	if isInTrans {
-		s.Status |= mysql.ServerStatusInTrans
+// SetStatus sets the session server status variable.
+// If flag is ture sets the input status in session status,
+// otherwise removes the input status.
+func (s *SessionVars) SetStatus(flag bool, status uint16) {
```

Comment:
I think rename as the following would be better
`SetStatusFlag(flag uint16, on bool)`